CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+project(PortableComponentModel)
+cmake_minimum_required(VERSION 3.17)
+set(CMAKE_CXX_STANDARD 17)
+
+# this is the type of build rules if the model is built as a standalone library
+add_library(
+        model
+        model/model.h
+        model/model.cpp
+)
+
+# this is the type of build rules if this was built as a part of EnergyPlus
+add_executable(
+        mock_energyplus
+        model/model.h
+        model/model.cpp
+        energyplus/Model.hh
+        energyplus/Model.cc
+        energyplus/EnergyPlus.hh
+        energyplus/EnergyPlus.cc
+)
+
```

Comment:
All the CMake rules live at the highest level.  I feel like this is appropriate given that for a component model going into E+, there won't be special rules for this one model, it will simply fit into the E+ rules.  And similar for a project utilizing the model as an FMU.